src/battle/other_battle_calculators.c

```diff
@@ -1452,3 +1463,104 @@ BOOL TerrainSeedShouldActivate(struct BattleStruct *sp, u16 heldItem) {
+    u16 rawPhysicalAttack;
+    u16 rawSpecialAttack;
+    u16 rawPhysicalDefense;
+    u16 rawSpecialDefense;
+
+    u16 tempPhysicalAttack = BattlePokemonParamGet(sp, attacker, BATTLE_MON_DATA_ATK, NULL) * StatBoostModifiers[attackerAttackstate][0] / StatBoostModifiers[attackerAttackstate][1];
+    u16 tempSpecialAttack = BattlePokemonParamGet(sp, attacker, BATTLE_MON_DATA_SPATK, NULL) * StatBoostModifiers[attackerSpecialAttackState][0] / StatBoostModifiers[attackerSpecialAttackState][1];
+
+    if (critical > 1) {
+        if (attackerAttackstate > 6) {
+            rawPhysicalAttack = attackerAttack * StatBoostModifiers[attackerAttackstate][0];
+            rawPhysicalAttack /= StatBoostModifiers[attackerAttackstate][1];
+        } else {
+            rawPhysicalAttack = attackerAttack;
+        }
+
+        if (defenderDefenseState < 6) {
+            rawPhysicalDefense = defenderDefense * StatBoostModifiers[defenderDefenseState][0];
+            rawPhysicalDefense /= StatBoostModifiers[defenderDefenseState][1];
+        } else {
+            rawPhysicalDefense = defenderDefense;
+        }
+
+        if (attackerSpecialAttackState > 6) {
+            rawSpecialAttack = attackerSpecialAttack * StatBoostModifiers[attackerSpecialAttackState][0];
+            rawSpecialAttack /= StatBoostModifiers[attackerSpecialAttackState][1];
+        } else {
+            rawSpecialAttack = attackerSpecialAttack;
+        }
+
+        if (defenderSpecialDefenseState < 6) {
+            rawSpecialDefense = defenderSpecialDefense * StatBoostModifiers[defenderSpecialDefenseState][0];
+            rawSpecialDefense /= StatBoostModifiers[defenderSpecialDefenseState][1];
+        } else {
+            rawSpecialDefense = defenderSpecialDefense;
+        }
+    } else {
+        rawPhysicalAttack = attackerAttack * StatBoostModifiers[attackerAttackstate][0];
+        rawPhysicalAttack /= StatBoostModifiers[attackerAttackstate][1];
+
+        rawPhysicalDefense = defenderDefense * StatBoostModifiers[defenderDefenseState][0];
+        rawPhysicalDefense /= StatBoostModifiers[defenderDefenseState][1];
+
+        rawSpecialAttack = attackerSpecialAttack * StatBoostModifiers[attackerSpecialAttackState][0];
+        rawSpecialAttack /= StatBoostModifiers[attackerSpecialAttackState][1];
+
+        rawSpecialDefense = defenderSpecialDefense * StatBoostModifiers[defenderSpecialDefenseState][0];
+        rawSpecialDefense /= StatBoostModifiers[defenderSpecialDefenseState][1];
+    }
+
+    if (movesplit == SPLIT_PHYSICAL) {
+        *equivalentAttack = rawPhysicalAttack;
+        *equivalentDefense = rawPhysicalDefense;
+    } else {
+        *equivalentAttack = rawSpecialAttack;
+        *equivalentDefense = rawSpecialDefense;
+    }
+
+    switch (moveno) {
+        case MOVE_PSYSHOCK:
+        case MOVE_PSYSTRIKE:
+        case MOVE_SECRET_SWORD:
+            *equivalentDefense = rawPhysicalDefense;
+            break;
```

Comment:
oh i guess ignore my other thing then

Reply:
so you mean to have stat-changing things here, but base power in the main damage calculator?